framework/src/org/checkerframework/framework/util/typeinference/TypeArgInferenceUtil.java

```diff
@@ -51,45 +51,43 @@
 public class TypeArgInferenceUtil {
 
     /**
-     * Takes an expression tree that must be either a MethodInovcationTree or a NewClassTree
-     * (constructor invocation) and returns the arguments to its formal parameters. An
-     * IllegalArgumentException will be thrown if it is neither
+     * Returns a list of boxed annotated types corresponding to the arguments in {@code
+     * methodInvocation}.
      *
-     * @param expression a MethodInvocationTree or a NewClassTree
-     * @return the list of arguments to Expression
+     * @param methodInvocation {@link MethodInvocationTree} or {@link NewClassTree}
+     * @param typeFactory type factory
+     * @return a list of boxed annotated types corresponding to the arguments in {@code
+     *     methodInvocation}.
      */
-    public static List<? extends ExpressionTree> expressionToArgTrees(
-            final ExpressionTree expression) {
+    public static List<AnnotatedTypeMirror> getArgumentTypes(
+            final ExpressionTree methodInvocation, final AnnotatedTypeFactory typeFactory) {
         final List<? extends ExpressionTree> argTrees;
-        if (expression.getKind() == Kind.METHOD_INVOCATION) {
-            argTrees = ((MethodInvocationTree) expression).getArguments();
 
-        } else if (expression.getKind() == Kind.NEW_CLASS) {
-            argTrees = ((NewClassTree) expression).getArguments();
+        if (methodInvocation.getKind() == Kind.METHOD_INVOCATION) {
+            argTrees = ((MethodInvocationTree) methodInvocation).getArguments();
 
-        } else {
-            argTrees = null;
-        }
+        } else if (methodInvocation.getKind() == Kind.NEW_CLASS) {
+            argTrees = ((NewClassTree) methodInvocation).getArguments();
 
-        if (argTrees == null) {
-            throw new IllegalArgumentException(
+        } else {
+            ErrorReporter.errorAbort(
                     "TypeArgumentInference.relationsFromMethodArguments:\n"
                             + "couldn't determine arguments from tree: "
-                            + expression);
+                            + methodInvocation);
+            throw new Error(); // dead code
         }
 
-        return argTrees;
-    }
-
-    /** Calls get annotated types on a List of trees using the given type factory. */
-    public static List<AnnotatedTypeMirror> treesToTypes(
-            final List<? extends ExpressionTree> argTrees, final AnnotatedTypeFactory typeFactory) {
-        final List<AnnotatedTypeMirror> argTypes = new ArrayList<>(argTrees.size());
+        final List<AnnotatedTypeMirror> argtypes = new ArrayList<>(argTrees.size());
```

Comment:
Tiny nit-pick: you go from `argTrees` with capital T to `argtypes` with lower-case T.